src/test/java/dog/pawbook/logic/commands/DeleteDogCommandTest.java

```diff
@@ -0,0 +1,114 @@
+package dog.pawbook.logic.commands;
+
+import static dog.pawbook.logic.commands.CommandTestUtil.assertCommandFailure;
+import static dog.pawbook.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static dog.pawbook.logic.commands.CommandTestUtil.showDogAtIndex;
+import static dog.pawbook.testutil.TypicalDogs.getTypicalAddressBook;
+import static dog.pawbook.testutil.TypicalIndexes.INDEX_FIRST_DOG;
+import static dog.pawbook.testutil.TypicalIndexes.INDEX_SECOND_DOG;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.commons.core.Messages;
+import dog.pawbook.commons.core.index.Index;
+import dog.pawbook.model.Model;
+import dog.pawbook.model.ModelManager;
+import dog.pawbook.model.UserPrefs;
+import dog.pawbook.model.managedentity.Entity;
+import javafx.util.Pair;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteCommand}.
+ */
+public class DeleteDogCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIdUnfilteredList_success() {
+        Pair<Integer, Entity> pair = model.getFilteredEntityList().get(0);
+        DeleteDogCommand deleteDogCommand = new DeleteDogCommand(Index.fromZeroBased(pair.getKey()));
```

Comment:
I only have myself to blame for this, maybe at least we can extract it into some form of a helper method in the util class? it isn't very straightforward that we are deleting the first dog since we are still using `getTypicalAddressBook` just from a different class.

Reply:
ok i'll look into it another time! let's merge this PR first and i ll continue from there onwards :)